build-conventions/src/main/kotlin/com/leinardi/forlago/ext/ProjectExt.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2024 Roberto Leinardi.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.leinardi.forlago.ext
+
+import org.gradle.api.Project
+import org.gradle.api.plugins.JavaPlugin
+import org.gradle.api.tasks.compile.JavaCompile
+import org.gradle.kotlin.dsl.withType
+
+fun Project.configureJavaCompile() {
+    plugins.withType<JavaPlugin> {
+        tasks.withType<JavaCompile>().configureEach {
+            options.compilerArgs.add("-Xlint:deprecation")
+            options.compilerArgs.add("-Xlint:unchecked")
+        }
+    }
+}
```

Comment:
why is this needed?

Reply:
Now I've seen you moved to a extension function, but why is only needed on `library conventions`?

Reply:
it is applied to the base android convention and to the base kotlin conventio, so it is applied to every project. The reason is this: https://discuss.gradle.org/t/basepluginconvention-deprecation-warning-for-archivefilename/46269/2

Reply:
True, I didnt see it on `android-conventions` 👍